pytorch_lightning/trainer/training_loop.py

```diff
@@ -471,13 +473,14 @@ def run_training_epoch(self):
 
         # run epoch
         for batch_idx, (batch, is_last_batch) in self.profiler.profile_iterable(
-                enumerate(_with_is_last(train_dataloader)), "get_train_batch"
+            enumerate(_with_is_last(train_dataloader)), "get_train_batch"
         ):
             # stop epoch if we limited the number of training batches
             if batch_idx >= self.num_training_batches:
                 break
 
             self.batch_idx = batch_idx
+            self.is_last_batch = is_last_batch
```

Comment:
let's pass this in as an argument to the function that needs it, like we do for "should_check_val" as example :)

Reply:
@awaelchli if we pass `limit_train_batches` to 0.x or 1x, I think `is_last_batch` will still be `False` even if we reach the last batch of `num_training_batches`. Is my assumption correct?

Reply:
oh you're right. ahh, this variable has a misleading name. your code that had num_training_batches was better, sorry :)